src/Matches/ViewFriendProfile.js

```diff
@@ -0,0 +1,134 @@
+import React from "react";
+import {withRouter, Link} from 'react-router-dom'
+import Header from '../Header/Header'
+
+class ViewFriendProfile extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        userId: this.props?.location?.state?.id,
+        friendId: this.props?.location?.state?.friendId,
+        name: "",
+        birthday: "",
+        age: 0,
+        bio: "",
+        gender: "Female",
+        genderPreference: "Female",
+        education: "",
+        interests: "",
+        error: "",
+      };
+      this.selectUserMessage = this.selectUserMessage.bind(this);
+    }
+    componentDidMount(){
+      const myRequest = new Request('http://127.0.0.1:5000/viewprofile', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({"userId": this.state.friendId}),
+        });
+    fetch(myRequest)
+        .then(response => response.json())
+        .then(res => 
+                this.setState({
+                    name: res.name,
+                    age: res.age, 
+                    bio: res.bio,
+                    gender: res.gender,
+                    genderPreference: res.genderPreference,
```

Comment:
should we show the gender preference of your friend? I feel that's confidential to the user itself and shouldn't be revealed to others since it's the user's own preference of selection.